cmd/artifactPrepareVersion.go

```diff
@@ -0,0 +1,351 @@
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	// open local .git repository
+	repository, err := openGit()
+	if err != nil {
+		log.Entry().WithError(err).Fatal("git repository required - none available")
+	}
+
+	err = runArtifactPrepareVersion(&config, telemetryData, commonPipelineEnvironment, nil, &c, repository, getGitWorktree)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("artifactPrepareVersion failed")
+	}
+	log.Entry().Info("SUCCESS")
+}
+
+var sshAgentAuth = ssh.NewSSHAgentAuth
+
+func runArtifactPrepareVersion(config *artifactPrepareVersionOptions, telemetryData *telemetry.CustomData, commonPipelineEnvironment *artifactPrepareVersionCommonPipelineEnvironment, artifact versioning.Artifact, runner execRunner, repository gitRepository, getWorktree func(gitRepository) (gitWorktree, error)) error {
+
+	telemetryData.Custom1Label = "buildTool"
+	telemetryData.Custom1 = config.BuildTool
+
+	// Options for artifact
+	artifactOpts := versioning.Options{
+		GlobalSettingsFile:  config.GlobalSettingsFile,
+		M2Path:              config.M2Path,
+		ProjectSettingsFile: config.ProjectSettingsFile,
+	}
+
+	var err error
+	if artifact == nil {
+		artifact, err = versioning.GetArtifact(config.BuildTool, config.FilePath, &artifactOpts, runner)
+		if err != nil {
+			return errors.Wrap(err, "failed to retrieve artifact")
+		}
+	}
+
+	versioningType := config.VersioningType
+
+	// support former groovy versioning template and translate into new options
+	if len(config.VersioningTemplate) > 0 {
+		versioningType, _, config.IncludeCommitID = templateCompatibility(config.VersioningTemplate)
+	}
+
+	version, err := artifact.GetVersion()
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve version")
+	}
+	log.Entry().Infof("Version before automatic versioning: %v", version)
+
+	gitCommit, err := getGitCommitID(repository)
+	if err != nil {
+		return err
+	}
+	gitCommitID := gitCommit.String()
+
+	newVersion := version
+
+	if versioningType == "cloud" {
+		versioningTempl, err := versioningTemplate(artifact.VersioningScheme())
+		if err != nil {
+			return errors.Wrapf(err, "failed to get versioning template for scheme '%v'", artifact.VersioningScheme())
+		}
+
+		t := time.Now()
+
+		newVersion, err = calculateNewVersion(versioningTempl, version, gitCommitID, config.IncludeCommitID, t)
+		if err != nil {
+			return errors.Wrap(err, "failed to calculate new version")
+		}
+
+		worktree, err := getWorktree(repository)
+		if err != nil {
+			return errors.Wrap(err, "failed to retrieve git worktree")
+		}
+
+		// opening repository does not seem to consider already existing files properly
+		// behavior in case we do not run initializeWorktree:
+		//   git.Add(".") will add the complete workspace instead of only changed files
```

Comment:
I have not found any either, thus the use of initialization.
Previously `git add .` worked correctly since the rest of the repo was unchanged. Here the behavior is different and `git add .` would add all files in the repo again ...